### PR TITLE
Hosting garnet as windows service

### PR DESCRIPTION
**Pull request by pradeepyadavmsft**
Created a package to host Garnet as a Windows service. 

It's similar to https://github.com/microsoft/garnet/pull/394/files but with a bit of refactoring to keep hosting as separate package.

### CODE REVIEW

File: hosting/Windows/Garnet.worker/Worker.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        /// <summary>
+        /// Triggered when the application host is performing a graceful shutdown.
+        /// </summary>
+        /// <param name="cancellationToken">Indicates that the shutdown process should no longer be graceful.</param>
+        public override async Task StopAsync(CancellationToken cancellationToken)
+        {
+            Dispose();
+            await base.StopAsync(cancellationToken).ConfigureAwait(false);
+            return;
```
Comment (PaulusParssinen) [inline hosting/Windows/Garnet.worker/Worker.cs:48]: nit:
```suggestion
```

File: Garnet.sln [hunk 2]
```diff
@@ -98,6 +98,12 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SampleModule", "playground\
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "GarnetJSON", "playground\GarnetJSON\GarnetJSON.csproj", "{2C8F1F5D-31E5-4D00-A46E-F3B1D9BC098F}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "hosting", "hosting", "{01823EA4-4446-4D66-B268-DFEE55951964}"
```
Comment (badrishc) [inline Garnet.sln:101]: why are hosting and Windows marked here as projects?

File: hosting/Windows/Garnet.worker/Worker.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        public override void Dispose()
+        {
+            if (_isDisposed)
+            {
+                return;
+            }
+            server?.Dispose();
+            _isDisposed = true;
+            GC.SuppressFinalize(this);
```
Comment (hez2010) [inline hosting/Windows/Garnet.worker/Worker.cs:59]: Why suppress the finalization while `Worker` doesn't have a finalizer at all?

File: Garnet.sln (hunk 2 above)
Comment (pradeepyadavmsft) [reply Garnet.sln:101]: image
In order to keep the structure like this.

File: hosting/Windows/Garnet.worker/Worker.cs (hunk 3 above)
Comment (pradeepyadavmsft) [reply hosting/Windows/Garnet.worker/Worker.cs:59]: Removed it keeping it consistent with the overall codebase.

**Review by badrishc — APPROVED** [review]

**Review by TalZaccai — APPROVED** [review]

**Pull request merged**